frontend/BoardGUI.py

```diff
@@ -37,29 +41,101 @@ def build_board(self):
+                # Move the piece to the new location
+                move_success = self.game_manager.move_piece(row, col)
+                if move_success:
+                    print(f"Piece moved to ({row}, {col})")
+                    # Remove the piece from the old position
+                    self.remove_piece(selected_row, selected_col)
+                    
+                    # Add the piece to the new position
+                    self.pieces_on_board[(row, col)] = self.game_manager.turn
+                    self.draw_board()  # Redraw board to show updated pieces
+                    
+                    # Check if the move forms a mill
+                    if self.game_manager.is_mill_formed(row, col):
+                        print("Mill formed! Remove opponent's piece.")
+                        self.game_manager.remove_opponent_piece(self.pieces_on_board)
+                    
+                    self.game_manager.end_turn()  # End turn after successful move
+                else:
+                    print(f"Invalid move to ({row}, {col})")
+            else:
+                print(f"Cell ({row}, {col}) is not adjacent or not empty")
+
+            # Clear the selected piece regardless of move success
+            self.game_manager.selected_piece = None
+    
+    
     def get_cell_clicked(self):
         current_cell = vec2(pg.mouse.get_pos()) // self.cell_size
         col, row = map(int, current_cell)
-        mouse_buttons = pg.mouse.get_pressed()  # Call this once and store the result
+        mouse_buttons = pg.mouse.get_pressed()
+        self.count += 1
+
+        # Handle opponent's piece removal if mill is formed
+        if self.game_manager.waiting_for_removal:
+            if mouse_buttons[0]:
+                self.handle_mill(row , col)
+            return current_cell
 
-        if mouse_buttons[0]:  # Left click
-            self.count += 1
-            # This part should be moved to the GameManager, even the draw_piece call
+        # Left click - Select or move piece
+        if mouse_buttons[0]:
             if not self.game_manager.placement_complete():
-                isPiecePlaced = self.game_manager.place_piece(row, col)
-                if isPiecePlaced == 1:
-                    self.draw_piece(
-                        self.game_manager.get_turn(), current_cell * self.cell_size
-                    )
-                    self.game_manager.end_turn()
-            return current_cell  # Return the current cell for left click
-
-        if mouse_buttons[2]:  # Right click
-            return None  # Return None on right click
-
-        return None  # Return None if no click
+                # Handle piece placement
+                self.game_manager.handle_piece_placement(row, col , self.pieces_on_board)
+                self.draw_board()
+                return current_cell  # Return the current cell for left click
+            else:
```

Comment:
Should check the GamePhase here instead.

Reply:
--